### ISSUE TITLE
EnumClassName = enum.Enum('EnumClassName', ...) should not be an invalid-name

### ISSUE THREAD

**Issue by gpshead** [PyCQA/pylint#1800]
pylint 1.8.1:
```
$ bin/pylint e.py
C:  7, 0: Constant name "MyThing" doesn't conform to UPPER_CASE naming style (invalid-name)
```
e.py:
```python
"""Testing enum.Enum top level name assignment."""

import enum

CONST = 13

MyThing = enum.Enum("MyThing", "foo bar spam")
```

The functional API of the enum module is widely used in addition to its subclassing behavior.  People rightfully want to declare top level enum's using ClassName style names rather than CLASS_NAME regardless of if they are using a class statement or an assignment to the result of the functional API.  Lets make an exception for enum.Enum top level assignments and not cause an invalid-name error when it also matches the CapWords class name style.  Continue allowing all CAPS constant style for such assignments as many people have already done that to work around this warning. (awkward, but valid)

**Comment by PCManticore** [comment]
Sounds good to me. This could be also applied to `namedtuple` and similar constructs.

**Comment by dmanikowski-reef** [comment]
Any custom config needed? I've just checked with pylint 2.0.0 and I am not able to reproduce it.

**Comment by gpshead** [comment]
I don't have any custom config, but I only tested with 1.8.1 as that is what pip gave me.  Also, agreed that this applies `namedtuple`.

**Comment by PCManticore** [comment]
This message is no longer emitted for non-const values with https://github.com/PyCQA/pylint/commit/3422e4adc718c60a467d66ab0dedf4a7032c4fc5
